### DISCUSSION TITLE
Livestream transcription/translation

### DISCUSSION THREAD

**Discussion by fortypercnt** [openai/whisper#225] [Show and tell]
I implemented a CLI tool that can use whisper directly on livestreams in ffmpeg, transcribing every couple of seconds. You can find it [here](https://github.com/fortypercnt/stream-translator). By default it uses [streamlink](https://github.com/streamlink/streamlink) to automatically get .m3u8 files for various streaming sites, but URLs can also be passed to ffmpeg directly.

**Comment by Mijawel** [comment]
I was wondering if your tool takes into account previous texts so that the chunks of transcribed text aren't disjointed?

**Comment by fortypercnt** [comment]
Unfortunately whisper does not provide a convenient way of dealing with real time audio. It does give the option to condition the model on previous text by giving a prompt, but in my limited testing the results were better without that. It often repeated itself and got stuck in loops. Mileage may vary when using windows longer than just a few seconds.
I might try to build something more sophisticated (e.g. based on sliding windows as described in https://github.com/openai/whisper/discussions/117#discussioncomment-3727051) when I have more time.

**Reply by jianfch** [reply]
`verbose=True` already prints the segments as they are being transcribed\translated.
so wouldn't it be more simple to just use `verbose=True`, instead of chopping the audio up even more and [printing](https://github.com/fortypercnt/stream-translator/blob/b058945d6583f9e63efb27d8f73811fb99e56c04/translator.py#L88) the results in blocking manner which makes results disjointed. 
if you want direct access to the data as it is being decoded, you can just add a hook to [add_segment()](https://github.com/openai/whisper/blob/0b1ba3d46ebf7fe6f953acfd8cad62a4f851b49f/whisper/transcribe.py#L143)

**Reply by fortypercnt** [reply]
The problem is with real time audio, there is only one segment for each call to transcribe/decode, which contains the last few seconds of audio captured from the live stream. Or maybe I am missing something.

**Reply by jianfch** [reply]
> there is only one segment for each call to transcribe/decode

oh i see, so wouldn't make a difference.
you can try during inference to fill in the padded portion with proceeding audio packets as it coming in. then shifting the window upon getting end-of-transcript token or reach 30 sec of unpadded audio. this way, you can mimic feeding a continuous stream of audio to avoid disjointed results.

**Comment by nyadla-sys** [comment]
Please see my project below, which uses the Whisper Tiny Tflite Model to implement audio streaming..
[Streaming using TFLite model](https://github.com/usefulsensors/openai-whisper/tree/main/stream)

**Comment by nyadla-sys** [comment]
Refer the below github repo for Android example, which uses whisper-tiny.en.tflite model 
https://github.com/nyadla-sys/whisper.tflite
